dataframe/benchmark_test.go

```diff
@@ -265,3 +288,591 @@ func BenchmarkDataFrame_Elem(b *testing.B) {
 		})
 	}
 }
+
+type joinColRowOrdering int
+
+const (
+	sorted joinColRowOrdering = iota
+	reversed
+	random
+)
+
+func shuffleSlice(slice []int) {
+	rand.Seed(time.Now().UnixNano())
+	rand.Shuffle(len(slice), func(i, j int) { slice[i], slice[j] = slice[j], slice[i] })
+}
+
+func generateJoinSeries(
+	colName string,
+	leftRowCount,
+	rightRowCount,
+	relationScaleLeftToRight,
+	relationScaleRightToLeft int,
+	joinRowOrdering joinColRowOrdering) (series.Series, series.Series) {
+
+	leftVals := make([]int, leftRowCount)
+	for i := 0; i < leftRowCount*relationScaleLeftToRight; i++ {
+		if i >= leftRowCount {
+			break
+		}
+		val := i / relationScaleLeftToRight
+		leftVals[i] = val + 1
+	}
+	rightVals := make([]int, rightRowCount)
+	for i := 0; i < rightRowCount*relationScaleRightToLeft; i++ {
+		if i >= rightRowCount {
+			break
+		}
+		val := i / relationScaleRightToLeft
+		if joinRowOrdering == reversed {
+			rightVals[i] = rightRowCount - val
+			continue
+		}
+		rightVals[i] = val + 1
+
+	}
+	if joinRowOrdering == random {
+		shuffleSlice(leftVals)
+		shuffleSlice(rightVals)
+	}
+	leftSeries := series.New(leftVals, series.Int, colName)
+	rightSeries := series.New(rightVals, series.Int, colName)
+
+	return leftSeries, rightSeries
+}
+
+func BenchmarkDataFrame_InnerJoinOptimized(b *testing.B) {
```

Comment:
nit: remove Optimized suffix for function test name here and in other places. Because it's the only benchmark function (there's no comparison with the old implementations anymore).

Reply:
You can also use this benchstat tool: https://pkg.go.dev/golang.org/x/perf/cmd/benchstat